db/l2db/l2db.go

```diff
@@ -147,58 +149,107 @@ func (l2db *L2DB) UpdateTxsInfo(txs []common.PoolL2Tx, batchNum common.BatchNum)
 	return nil
 }
 
-// NewPoolL2TxWriteFromPoolL2Tx creates a new PoolL2TxWrite from a PoolL2Tx
-func NewPoolL2TxWriteFromPoolL2Tx(tx *common.PoolL2Tx) *PoolL2TxWrite {
-	// transform tx from *common.PoolL2Tx to PoolL2TxWrite
-	insertTx := &PoolL2TxWrite{
-		TxID:      tx.TxID,
-		FromIdx:   tx.FromIdx,
-		TokenID:   tx.TokenID,
-		Amount:    tx.Amount,
-		Fee:       tx.Fee,
-		Nonce:     tx.Nonce,
-		State:     common.PoolL2TxStatePending,
-		Signature: tx.Signature,
-		RqAmount:  tx.RqAmount,
-		Type:      tx.Type,
-	}
-	if tx.ToIdx != 0 {
-		insertTx.ToIdx = &tx.ToIdx
+// AddTxTest inserts a tx into the L2DB, without security checks. This is useful for test purposes,
+func (l2db *L2DB) AddTxTest(tx *common.PoolL2Tx) error {
+	// Add tx without checking if pool is full
+	_, err := l2db.addTx(tx, false)
+	return err
+}
+
+func (l2db *L2DB) addTx(tx *common.PoolL2Tx, checkPoolIsFull bool) (sql.Result, error) {
+	// Prepare extra DB fields and nullables
+	// AmountFloat
+	f := new(big.Float).SetInt((*big.Int)(tx.Amount))
```

Comment:
Isn't tx.Amount already have *big.Int type?

Reply:
yes, this is used to store the amount as float in the DB. There is a column in the table `amount_f` that requires it